lib/src/common/disposable.dart

```diff
@@ -69,7 +69,7 @@ class ManagedDisposer implements _Disposable {
 
     var disposeFuture = _disposer != null
         ? (_disposer() ?? new Future.value())
-        : new Future.value();
+        : new Future<dynamic>.value();
```

Comment:
Kinda sucks that we have to to this with these options :(

Buuuuut I'm still in favor of it if it means we catch RTEs statically.